snippets/csharp/keywords/GenericWhereConstraints.cs

```diff
@@ -46,15 +46,13 @@ public interface IMyInterface { }
     namespace CodeExample
     {
         class Dictionary<TKey, TVal>
-            where TKey : IComparable, IEnumerable
+            where TKey : IComparable<TKey>
```

Comment:
I'm curious why you dropped `IEnumerable`?

Reply:
My confusion was: Why to enumerate through a key in the dictionary context? Quite a specific use case. But yes, of course, that might be. But then I was a little bit against non-generic interface in the example and `TKey : IEnumerable<TKey>` looked even more fancier. Thus, I gave up.

I would rather expect `class Dictionary<TKey, TVal> : IEnumerable<TVal>`, by the way.
If we want to show several interfaces in one `where` constraint, what about
```csharp
where TKey : IComparable<T>, IEquatable<T>
```
Or it's better not to mix those two interfaces without care?

Reply:
Thanks for the extra explanation.

Removing `IEnumerable` is the right choice. It isn't enlightening the discussion, and is not likely a good constraint on the key type.

I'll :shipit: now.